src/eip7594/recovery.c

```diff
@@ -188,18 +187,19 @@ static bool is_in_array(const uint64_t *arr, size_t arr_size, uint64_t value) {
 }
 
 /**
- * Given a dataset with up to half the entries missing, return the reconstructed original. Assumes
- * that the inverse FFT of the original data has the upper half of its values equal to zero.
+ * Given a set of cells with up to half the entries missing, return the reconstructed
+ * original. Assumes that the inverse FFT of the original data has the upper half of its values
+ * equal to zero.
  *
- * @param[out]  reconstructed_data_out   Preallocated array for recovered cells
- * @param[in]   cell_indices             The cell indices you have
- * @param[in]   num_cells                The number of cells that you have
- * @param[in]   cells                    The cells that you have
+ * @param[out]  reconstructed_data_out   Array of size FIELD_ELEMENTS_PER_EXT_BLOB to recover cells
+ * @param[in]   cell_indices             An array with the avaibable cell indices we have
+ * @param[in]   num_cells                The size of the `cell_indices` array
+ * @param[in]   cells                    An array of size FIELD_ELEMENTS_PER_EXT_BLOB with the cells
  * @param[in]   s                        The trusted setup
  *
- * @remark `recovered` and `cells` can point to the same memory.
- * @remark The array of cells must be 2n length and in the correct order.
- * @remark Missing cells should be equal to FR_NULL.
+ * @remark `reconstructed_data_out` and `cells` can point to the same memory.
+ * @remark The array `cells` must be in the correct order (according to cell_indices)
```

Comment:
This remark is missing punctuation:
```suggestion
 * @remark The array `cells` must be in the correct order (according to cell_indices).
```